Polyfills/Canvas/Source/Context.cpp

```diff
@@ -0,0 +1,694 @@
+                InstanceMethod("stroke", &Context::Stroke),
+                InstanceMethod("fill", &Context::Fill),
+                InstanceMethod("drawImage", &Context::DrawImage),
+                InstanceMethod("getImageData", &Context::GetImageData),
+                InstanceMethod("setLineDash", &Context::SetLineDash),
+                InstanceMethod("fillText", &Context::FillText),
+                InstanceMethod("strokeText", &Context::StrokeText),
+                InstanceMethod("createLinearGradient", &Context::CreateLinearGradient),
+                InstanceMethod("setTransform", &Context::SetTransform),
+                InstanceAccessor("lineJoin", &Context::GetLineJoin, &Context::SetLineJoin),
+                InstanceAccessor("miterLimit", &Context::GetMiterLimit, &Context::SetMiterLimit),
+                InstanceAccessor("font", &Context::GetFont, &Context::SetFont),
+                InstanceAccessor("strokeStyle", &Context::GetStrokeStyle, &Context::SetStrokeStyle),
+                InstanceAccessor("fillStyle", &Context::GetFillStyle, &Context::SetFillStyle),
+                InstanceAccessor("globalAlpha", &Context::GetGlobalAlpha, &Context::SetGlobalAlpha),
+                InstanceAccessor("shadowColor", &Context::GetShadowColor, &Context::SetShadowColor),
+                InstanceAccessor("shadowBlur", &Context::GetShadowBlur, &Context::SetShadowBlur),
+                InstanceAccessor("shadowOffsetX", &Context::GetShadowOffsetX, &Context::SetShadowOffsetX),
+                InstanceAccessor("shadowOffsetY", &Context::GetShadowOffsetY, &Context::SetShadowOffsetY),
+                InstanceAccessor("lineWidth", &Context::GetLineWidth, &Context::SetLineWidth),
+                InstanceAccessor("canvas", &Context::GetCanvas, nullptr)
+            });
+        return func.New({ Napi::External<NativeCanvas>::New(env, canvas)});
+    }
+
+    Context::Context(const Napi::CallbackInfo& info)
+        : Napi::ObjectWrap<Context>{info}
+        , m_canvas{info[0].As<Napi::External<NativeCanvas>>().Data()}
+        , m_nvg{nvgCreate(1)}
+        , m_graphicsImpl{Babylon::GraphicsImpl::GetFromJavaScript(info.Env())}
+        , m_cancellationSource{std::make_shared<arcana::cancellation_source>()}
+        , m_runtimeScheduler{Babylon::JsRuntime::GetFromJavaScript(info.Env())}
+    {
+        for (auto& font : NativeCanvas::fontsInfos)
+        {
+            m_fonts[font.first] = nvgCreateFontMem(m_nvg, font.first.c_str(), font.second.data(), font.second.size(), 0);
+        }
+        BeginFrame();
+    }
+
+    Context::~Context()
+    {
+        nvgDelete(m_nvg);
+        m_cancellationSource->cancel();
+    }
+
+    NVGcolor StringToColor(const std::string& colorString)
+    {
+        std::string str = colorString;
+        std::transform(str.begin(), str.end(), str.begin(),
+            [](unsigned char c) { return std::tolower(c); });
+
+        static const std::map<std::string, uint32_t> webColors = {
```

Comment:
consider putting this in a separate header file

Reply:
yes, next PR.